platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java

```diff
@@ -44,12 +51,18 @@
 
     private CRL crl = null;
 
+    private ScheduledExecutorService scheduler;
+
+    private Lock crlLock = new ReentrantLock();
+
+    private CrlRefresh refresh = new CrlRefresh();
+
     /**
      * Constructor method. Reads encryption.properties and sets CRL location
      */
     public CrlChecker() {
-        Properties encryptionProperties = loadProperties(encryptionPropertiesLocation);
-        setCrlLocation(encryptionProperties.getProperty(CRL_PROPERTY_KEY));
+        scheduler = Executors.newScheduledThreadPool(1);
+        scheduler.scheduleWithFixedDelay(refresh, 0, 1, TimeUnit.HOURS);
```

Comment:
I debated about this as well. There isn't any configuration in this bundle currently and I feel like it would just be configuration for the sake of configuration.  I'm thinking of the scenario where you know the CRL updates every 24 hours. If you schedule it to every 24 hours, it likely won't align because the timer will start when the instance is created. I think I see more value in a forced update after some set interval (like once a day?) in case the integrator just sets the nextUpdate to some point in the future and actually updates the CRL prior to nextUpdate.

The RFC I've been reading for reference: https://www.ietf.org/rfc/rfc5280.txt